test/LEX/live.py

```diff
@@ -40,15 +42,24 @@
 if not lex:
     test.skip_test('No lex or flex found; skipping test.\n')
 
+tools = "'default'"
+if sys.platform == 'win32':
+    # make sure mingw is installed on win32
+    if not test.where_is('gcc'):
+        test.skip_test('No mingw on windows; skipping test.\n')
+    # lex on win32 has a dependencies on mingw for unix headers
+    # so add it as a tool to the environment.
+    tools += ", 'mingw'"
```

Comment:
Should you still need this with the logic to check mingw paths in lex tool above?

Reply:
The Unix headers won't be part of the default include path unless mingw is added to the environment on windows. I don't know exactly how lex works, but this test generates some code which needs Unix headers which mingw brings to windows.

Reply:
lex takes an input file which specifies tokens to be tokenized as part a parser (often build with yacc/bison/byacc).  It generates a .c and a .h file (and potentially other debug output files).

It should be able to be compiled by MSVC on windows though. 
Does it fail with msvc?

Reply:
Yeah, it fails to compile because it can't find unistd.h with just the default tools on windows.

Do you know if it possible to use lex to generate code that doesn't include unix stuff? I will take a look when I get a chance.

Reply:
It should be able to. Which lex are you picking up in the tests?
Maybe try:
https://sourceforge.net/projects/winflexbison/

Which is in chocolatey.

Basically the skeleton file used by flex/lex will have unistd if flex is built in an environment which has it. so likely if you're using mingw's lex it will depend on unistd. Also you can manually specify the skeleton file. (Using -Sskeleton)

Reply:
Also in the past I've seen  sed scripts to rip the #include unistd.h out.

Reply:
flex at least has an option --nounistd

Reply:
And this for winflexbison:  https://github.com/lexxmark/winflexbison/commit/daca9cadd0f7139a9875f417ddf8872cfb1f338f

Reply:
ok added a check for 'win_flex' binary and used the --nounistd on windows as default flag. That flag also works with flex in mingw.